r4/smartconfig/modules/models/model_config.bal

```diff
@@ -0,0 +1,60 @@
+
+// Copyright (c) 2023, WSO2 LLC. (http://www.wso2.com).
+
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+# Smart configuration record
+#
+# + discoveryEndpoint - Smart configuration discoveryEndpoint
+# + smartConfiguration - Smart configuration
+public type configs record {
+    string discoveryEndpoint?;
+    configSmartConfiguration smartConfiguration?;
+};
+
+# Smart configuration record
+#
+# + issuer - Smart configuration issuer  
+# + jwksUri - Smart configuration jwks_uri  
+# + authorizationEndpoint - Smart configuration authorization_endpoint  
+# + grantTypesSupported - Smart configuration grant_type_supported  
+# + tokenEndpoint - Smart configuration token_endpoint  
+# + tokenEndpointAuthMethodsSupported - Smart configuration token_endpoint_auth_methods_supported  
+# + tokenEndpointAuthSigningAlgValuesSupported - Smart configuration token endpoint auth signing alg values supported
+# + registrationEndpoint - Smart configuration registration_endpoint  
+# + scopesSupported - Smart configuration scopes_supported  
+# + responseTypesSupported - Smart configuration response_type_supported  
+# + managementEndpoint - Smart configuration management_endpoint  
+# + introspectionEndpoint - Smart configuration introspection_endpoint  
+# + revocationEndpoint - Smart configuration revocation_endpoint  
+# + capabilities - Smart configuration capabilities  
+# + codeChallengeMethodsSupported - Smart configuration code_challenge_methods_supported
+public type configSmartConfiguration record {
```

Comment:
If these are the only fiedls that can be in `configSmartConfiguration` its better to make this a closed record. ie use `record {| |}`.

Reply:
Fixed in https://github.com/ballerina-platform/module-ballerinax-health.fhir.templates/commit/8693536cd1ee02c420a6325c46a80bc44337ed91